src/web/mjs/connectors/MangaDex.mjs

```diff
@@ -47,66 +47,59 @@ export default class MangaDex extends Connector {
-            }
-        }
-        return mangaList;
-    }
+        let data1 = await this._getMangasFromPages(0, 99);
+        mangaList = [...mangaList, ...data1.data];
+        let nextAt = data1.nextAt;
 
-    async _getMangasFromPage(page, query) {
-        if(page > 99) {
-            // NOTE: Current limit is 10000 entries (elastic search limit)
-            //       => Do not throw an error but return gracefully instead, so at least the partial list is shown
-            return [];
+        for (let i = 1; i <= data1.total / 10000; i += 1) {
+            let data2 = await this._getMangasFromPages(0, 0, nextAt);
+            mangaList = [...mangaList, ...data2.data.slice(1)];
+            let pages = Math.min(Math.floor(data2.total / 100), 99);
+            if (pages > 0) {
+                let data3 = await this._getMangasFromPages(1, pages, nextAt);
+                mangaList = [...mangaList, ...data3.data];
+                nextAt = data3.nextAt;
+            }
         }
-        await this.wait(this.throttleGlobal);
-        const uri = new URL('/manga', this.api);
-        uri.searchParams.set('limit', 100);
-        uri.searchParams.set('offset', 100 * page);
-        uri.searchParams.set('publicationDemographic[]', query.publicationDemographic);
-        uri.searchParams.set('order[createdAt]', query.order);
-        uri.searchParams.append('contentRating[]', 'safe');
-        uri.searchParams.append('contentRating[]', 'suggestive');
-        uri.searchParams.append('contentRating[]', 'erotica');
-        uri.searchParams.append('contentRating[]', 'pornographic');
-        const request = new Request(uri, this.requestOptions);
-        const {data} = await this.fetchJSON(request, 3);
-        return !data ? [] : data.map(result => {
-            const title = document.createElement('div');
-            title.innerHTML = result.attributes.title.en || Object.values(result.attributes.title).shift();
+        return mangaList.map(ele => {
             return {
-                id: result.id,
-                title: title.textContent.trim()
+                id: ele.id,
+                title: (ele.attributes.title.en || Object.values(ele.attributes.title).shift()).trim()
             };
         });
     }
 
+    async _getMangasFromPages(start, pages, nextAt) {
+        let tmp = [];
+        for (let page = start; page <= pages; page += 1) {
+            const uri = new URL('/manga', this.api);
+            uri.searchParams.set('limit', 100);
+            uri.searchParams.set('offset', 100 * page);
+            uri.searchParams.set('order[createdAt]', 'asc');
+            if (nextAt) uri.searchParams.set('createdAtSince', nextAt);
+            const request = new Request(uri);
+            let data3 = await this.fetchJSON(request, 3);
+            await this.wait(this.throttleGlobal);
+            tmp = [...tmp, ...data3.data];
+            if (page == pages) {
```

Comment:
Condition is redundant, it is expected that `page == pages` is always the case after the `for` loop